lib/std/heap/general_purpose_allocator.zig

```diff
@@ -231,7 +231,12 @@ pub fn GeneralPurposeAllocator(comptime config: Config) type {
                 const start_ptr = @ptrCast([*]u8, bucket) + bucketStackFramesStart(size_class);
                 const addr = start_ptr + one_trace_size * traces_per_slot * slot_index +
                     @enumToInt(trace_kind) * @as(usize, one_trace_size);
-                return @ptrCast(*[stack_n]usize, @alignCast(@alignOf(usize), addr));
+                if (stack_n > 0) {
```

Comment:
Nit, you can save one level of indentation by handling the exceptional case before the general one:

```zig
if (stack_n == 0) return &[_]usize{};
return @ptrCast(*[stack_n]usize, @alignCast(@alignOf(usize), addr));
```

Reply:
I guess you could do that, but it would have to look more like

```rust
comptime var empty_arr = [_]usize{};
if (stack_n == 0) return &empty_arr;
return @ptrCast(*[stack_n]usize, @alignCast(@alignOf(usize), addr));
```

Because in your fix, it doesn't work since `&[_]usize{}` is a const pointer.

Maybe this would be better too, since we won't need 2 return statements:

```rust
comptime var empty_arr = [_]usize{};
return if (stack_n != 0) 
    @ptrCast(*[stack_n]usize, @alignCast(@alignOf(usize), addr))
else
    &empty_arr;
```

However, the downside to that is that the compiler will generate `empty_arr` even in the case where `stack_n` is not zero. But due to lazy evaluation this is maybe ok?

Reply:
> Because in your fix, it doesn't work since &[_]usize{} is a const pointer.

Welp, I didn't expect the constness to matter there.
You're basically forced to return a local pointer to a zero sized variable, it's not that nice :\

> But due to lazy evaluation this is maybe ok?

It's an empty array, its size won't matter much/at all.